src/vehicle.cpp

```diff
@@ -4693,7 +4794,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+
+            if( is_body_collision && parts[ret.part].enabled &&
+                part_info( ret.part ).has_flag( "CHAINSAW" ) ) {
+                dam += damage_from_chainsaw( ret.part, critter->get_material() );
```

Comment:
It would be better to pass the entire critter reference here rather than just the material.

Currently all you use is the material and only the first material, but later on particularly tough critters made of soft materials (bone) could cause more self-damage to saws, multi-material critters could be handled, blood spills could be handled etc.
For all of the above, an entire critter reference would be more useful than just the material.